action_server/src/action_server/actions/hand_over.py

```diff
@@ -79,13 +78,20 @@ def _configure(self, robot, config):
                     self._config_result.required_context['object'] = config.semantics['object']
                 elif 'object' in config.context and 'type' in config.context['object']:
                     self._config_result.required_context['object'] = {'type': config.context['object']['type']}
+
             if 'source-location' in config.semantics:
                 self._config_result.required_context['source-location'] = config.semantics['source-location']
             if 'location' in config.context and 'id' in config.context['location']:
                 self._config_result.required_context['source-location'] = config.context['location']['id']
+
+            # If necessary request an object from the given category
+            #ToDo this needs the grammar to make a difference between CATEGORY and NAMED_OBJECT
+            if 'category' in config.semantics and 'category' in config.semantics['category']:
```

Comment:
Why would `category` be in semantics["category"]?

Reply:
Because the command contains something like 'bring me a drink' and then 'drink' is a category. Semantics contains the action description.

Reply:
I understand that `category` can be in semantics, but not why `category` is in `semantics["category"]`.

Reply:
You are right it is unnecessary and it has therefore been removed.